bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -485,4 +491,17 @@ public static String guessContentTypeFromData(byte[] data) {
         return contentType;
     }
 
+    /**
+     * Check whether the content data is a JPEG file checking file start and end bytes.
+     * {@link URLConnection#guessContentTypeFromStream(InputStream)} is wrong for some JPEG files.
+     *
+     * @see https://en.wikipedia.org/wiki/JPEG#Syntax_and_structure
```

Comment:
To me, this reads like it would be safe to additionally check for `data[2] == (byte) 0xFF`. I currently can't find a counter example what it would be absolutely required, but just to be on the safe side...